### PR TITLE
Dev secrets

### LINKED ISSUE #568: Support for developer secrets
**Is your feature request related to a problem? Please describe.**
Currently it is not possible to define additional secrets that should only be available when running in development mode.

**Describe the solution you'd like**
It would be helpfull when it's possible to define expandable secrets in the okteto manifest that can be mounted as either a file or as environment variable when running in development mode.

**Describe alternatives you've considered**
This is currently not possible and manual actions are required a.t.m.

**Additional context**
This can be usefull for setting specific information like developer ssh keys, db credentials as environment variables etc.

### PR DESCRIPTION
**Pull request by pchico83**
Fixes #568

### CODE REVIEW

File: cmd/up.go [hunk 1]
```diff
@@ -481,7 +481,7 @@ func (up *UpContext) startLocalSyncthing() error {
 	if err := up.Sy.WaitForPing(up.Context, false); err != nil {
 		return errors.UserError{
 			E:    fmt.Errorf("Failed to connect to the synchronization service"),
-			Hint: fmt.Sprintf("If you are using a non-root container, set the securityContext.runAsUser, securityContext.runAsGroup and securityContext.fsGroup fields in your Okteto manifest (https://okteto.com/docs/reference/manifest/index.html#securityContext-object-optional).\n    Run 'okteto down -v' to reset the synchronization service and try again."),
+			Hint: fmt.Sprintf("Check your development environment logs for errors: 'kubectl logs %s'\n    If you are using a non-root container, set the securityContext.runAsUser, securityContext.runAsGroup and securityContext.fsGroup fields in your Okteto manifest (https://okteto.com/docs/reference/manifest/index.html#securityContext-object-optional).\n    If you are using secrets, check that your container can write to the destination path of your secrets.\n    Finally, run 'okteto down -v' to reset the synchronization service and try again.", up.Pod),
```
Comment (rberrelleza) [inline cmd/up.go]:
```suggestion
			Hint: fmt.Sprintf("Check your development environment logs for errors: 'kubectl logs %s'.\n    If you are using a non-root container, set the securityContext.runAsUser, securityContext.runAsGroup and securityContext.fsGroup fields in your Okteto manifest (https://okteto.com/docs/reference/manifest/index.html#securityContext-object-optional).\n    If you are using secrets, check that your container can write to the destination path of your secrets.\n    Finally, run 'okteto down -v' to reset the synchronization service and try again.", up.Pod),
```

File: pkg/k8s/deployments/translate.go [hunk 2]
```diff
@@ -427,9 +439,59 @@ func TranslateOktetoSyncSecret(spec *apiv1.PodSpec, name string) {
 		Name: oktetoSyncSecretVolume,
 		VolumeSource: apiv1.VolumeSource{
 			Secret: &apiv1.SecretVolumeSource{
-				SecretName: fmt.Sprintf(oktetoSyncSecretTemplate, name),
+				SecretName: fmt.Sprintf(oktetoSecretTemplate, name),
+				Items: []apiv1.KeyToPath{
```
Comment (rberrelleza) [inline pkg/k8s/deployments/translate.go]: is there a risk that these get mixed with the dev secrets if they have the same name?

File: pkg/k8s/deployments/translate.go [hunk 3]
```diff
@@ -427,9 +439,59 @@ func TranslateOktetoSyncSecret(spec *apiv1.PodSpec, name string) {
...
+				SecretName: fmt.Sprintf(oktetoSecretTemplate, secret),
+				Items:      []apiv1.KeyToPath{},
+			},
+		},
+	}
+	for i, s := range secrets {
+		v.VolumeSource.Secret.Items = append(
+			v.VolumeSource.Secret.Items,
+			apiv1.KeyToPath{
+				Key:  filepath.Base(s.RemotePath),
```
Comment (rberrelleza) [inline pkg/k8s/deployments/translate.go]: why is key and path the same thing?

File: pkg/model/dev.go [hunk 4]
```diff
@@ -111,6 +112,13 @@ type EnvVar struct {
 	Value string
 }
 
+// Secret represents a development secret
+type Secret struct {
+	LocalPath  string
+	RemotePath string
+	Mode       int32
```
Comment (rberrelleza) [inline pkg/model/dev.go]: what's the mode for?

File: pkg/model/serializer_test.go [hunk 5]
```diff
@@ -203,6 +206,100 @@ func TestEnvVarMashalling(t *testing.T) {
...
+			nil,
+		},
+		{
+			"variables",
+			"$TEST_HOME:/remote",
+			&Secret{LocalPath: file.Name(), RemotePath: "/remote", Mode: 420},
+			nil,
+		},
+		{
+			"too-sort",
```
Comment (rberrelleza) [inline pkg/model/serializer_test.go]:
```suggestion
			"too-short",
```

File: pkg/model/serializer.go [hunk 6]
```diff
@@ -34,6 +34,47 @@ func (e EnvVar) MarshalYAML() (interface{}, error) {
...
+	var raw string
+	err := unmarshal(&raw)
+	if err != nil {
+		return err
+	}
+
+	rawExpanded := os.ExpandEnv(raw)
+	parts := strings.Split(rawExpanded, ":")
+	if len(parts) < 2 || len(parts) > 3 {
+		return fmt.Errorf("secrets must follow the syntax 'LOCAL_PATH:REMOTE_PATH:MODE'")
```
Comment (rberrelleza) [inline pkg/model/serializer.go]: would it be worth adding a default mode? (`400`)

File: pkg/model/serializer.go [hunk 7]
```diff
@@ -34,6 +34,47 @@ func (e EnvVar) MarshalYAML() (interface{}, error) {
 	return e.Name + "=" + e.Value, nil
 }
 
+// UnmarshalYAML Implements the Unmarshaler interface of the yaml pkg.
+func (s *Secret) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
Comment (rberrelleza) [inline pkg/model/serializer.go]: do we also need to check that every secret is unique?

File: pkg/k8s/deployments/translate.go (hunk 2 above)
Comment (pchico83) [reply pkg/k8s/deployments/translate.go]: Good catch, I will add a prefix to the dev secrets ones to avoid collisions

**Pull request merged**